src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/histogram.js

```diff
@@ -0,0 +1,533 @@
+loadScript('https://d3js.org/d3.v4.min.js');
+loadStyle('https://fontlibrary.org/face/dejavu-sans-mono')
+
+let shortcuts = {
+    zoomIn  : (e) => ((e.ctrlKey || e.metaKey) && e.key === 'z'),
+    showAll : (e) => ((e.ctrlKey || e.metaKey) && e.key === 'a')
+}
+
+const LABEL_STYLE            = "font-family: DejaVuSansMonoBook; font-size: 10px;";
+const X_AXIS_LABEL_WIDTH     = 30;
+const ANIMATION_DURATION     = 1000;
+const LINEAR_SCALE           = "linear";
+const LIGHT_PLOT_COLOR       = "#00E890";
+const DARK_PLOT_COLOR        = "#E0A63B";
+const DEFAULT_NUMBER_OF_BINS = 10;
+const BUTTON_HEIGHT          = 25;
+
+
+/**
+ * A d3.js histogram visualization.
+ *
+ *
+ * Data format (json):
+ {
+   "axis" : {
+      "x" : { "label" : "x-axis label", "scale" : "linear" },
+      "y" : { "label" : "y-axis label", "scale" : "logarithmic" },
+   },
+   "focus" { "x" : 1.7, "y" : 2.1, "zoom" : 3.0 },
+   "color" :  "rgb(1.0,0.0,0.0)" },
+   "bins" : 10,
+   "data" : [
+     "values" : [0.1, 0.2, 0.1, 0.15, 0.7],
+   ]
+ }
+ */
+class Histogram extends Visualization {
+    static inputType = "Any"
+    static label     = "Histogram (JS)"
```

Comment:
The labels should not really have info what language this is written in - users dont care aobut it